packages/reactive-element/src/reactive-element.ts

```diff
@@ -42,7 +42,7 @@ if (DEV_MODE) {
 
   // Issue platform support warning.
   if (
-    (window as LitExtraGlobals).ShadyDOM?.inUse &&
+    window.ShadyDOM?.inUse &&
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     (globalThis as any)['reactiveElementPlatformSupport'] === undefined
```

Comment:
I think getting `reactiveElementPlatformSupport`, etc. off of the global shouldn't require `as any` anymore?

Reply:
Seems like it, did you try?

Reply:
Not yet. I'll try and then remove these if it works.

Reply:
There's a bit more fallout than I expected around removing these casts, so I've put them in a separate PR with this PR as the base branch: #2043. Should I go ahead and merge this one or wait until the other one is ready too?

Reply:
Internal builds seem to be fine with that new PR, so I'll just wait until a call is made there.

Reply:
Based on our meeting this morning, I'm going to merge this PR before #2043 since it's already approved. Before that, I'll merge `main` here again to double check that nothing that would affect the tests has changed in the meantime.